File: 프로그래머스/2/42584. 주식가격/주식가격.java
import java.util.*;
class Solution {
    public int[] solution(int[] prices) {
        int[] answer = {};
        answer = new int[prices.length];
        Stack<Integer> stack = new Stack<>();
        Stack<Integer> idx_stack = new Stack<>();
        stack.push(prices[0]);
        idx_stack.push(0);
        for(int i = 1; i < prices.length; i++) {
        	if(!stack.isEmpty()) {
                while(!stack.isEmpty() && stack.peek() > prices[i])
                {
                    answer[idx_stack.peek()] = i-idx_stack.peek();
        		    stack.pop();
        		    idx_stack.pop();
                }
        	}
            stack.push(prices[i]);
        	idx_stack.push(i);
        }
        while(!stack.isEmpty()) {
        	answer[idx_stack.peek()] = answer.length - idx_stack.peek() - 1;
        	stack.pop();
        	idx_stack.pop();
        }
        return answer;
    }
}

File: 프로그래머스/2/42584. 주식가격/주식가격.py
def solution(prices):
    # 이거 스택이고... 내가 봤을때 idx에 따라와 stack의 길이에 따라서 장난 치는 것 같아.. 이거 말고는 말이 안되.
    
    answer = [0 for _ in range(len(prices))]
    stack = []
    for i in range(len(prices)):
        time = 0
        for j in range(i+1, len(prices)):
            time += 1
            if prices[i] > prices[j]:
                break
        answer[i] = time
    return answer